### PR TITLE
Added edit profile functionality

### PR DESCRIPTION
**Pull request by colinchoix**

### CODE REVIEW

File: users/views.py [hunk 1]
```diff
@@ -125,3 +132,32 @@ def friends(request, author_id):
...
+
+    if request.method == "POST":
+        form = EditProfileForm(request.POST)
+
+        if form.is_valid():
+            try:
+                request.user.username = form.cleaned_data["username"]
+                request.user.save()
+                return redirect("profile", author_id=you.pk)
+            except:
```
Comment (petelliott) [inline users/views.py]: don't use bare excepts, figure out what actual errors you expect to catch

File: users/templates/users/profile.html [hunk 2]
```diff
@@ -32,6 +32,23 @@ <h2>{{ author.user.username }}</h2>
 
 </div>
 <div class="profile-body">
+
+  <div id="editProfileForm" class="edit-profile">
+    <div class="edit-profile-content">
```
Comment (petelliott) [inline users/templates/users/profile.html:36]: why the double div?

File: users/views.py [hunk 3]
```diff
@@ -125,3 +132,32 @@ def friends(request, author_id):
                     {"author": author,
                      "followers": author.get_followers(),
                      "following": author.get_following()})
+
+def editProfile(request, author_id):
```
Comment (petelliott) [inline users/views.py]: there is no reason to ever GET /author/id/edit, so use the `@require_POST` decorator from `django.views.decorators.http`, and get rid of the `if request.method == "POST"`

**Review by petelliott — CHANGES_REQUESTED** [review]
looks really good so far!

File: users/templates/users/profile.html (hunk 2 above)
Comment (colinchoix) [reply users/templates/users/profile.html:36]: Intentional. The outer div is what dims the background when the popup form is open and is also what causes the form to close if you click outside the form.
The inner div is the actual form

**Review by petelliott — DISMISSED** [review]

**Comment by colinchoix** [comment]
Moved friend requests to Friends page. 
A red badge will show up in your profile beside Friends letting you know how many friend requests you have

**Comment by petelliott** [comment]
> Moved friend requests to Friends page.
> A red badge will show up in your profile beside Friends letting you know how many friend requests you have

~this doesn't seem to work for me. also we should keep friend requests as an explicit concept, because that is required by the spec~

Edit: nvm, I tested it wrong.

File: users/templates/users/friends.html [hunk 4]
```diff
@@ -4,23 +4,42 @@
...
-        <ul>
-        {% for f in following %}
-            <li><a href={% url 'profile' f.pk %}>{{f.user.username}}</a></li>
-        {% endfor %}
-        </ul>
+    <div class="content">
+        <div class="friends">            
+            <ul>
+            {% if author and author.user == user and freqs|length  > 0 %}
+                <li>Friend Requests</li>
```
Comment (petelliott) [inline users/templates/users/friends.html]: put friend requests in a separate list, currently it looks weird having '--' as the sperator, and titles as list elements

**Review by petelliott — CHANGES_REQUESTED** [review]

**Review by petelliott — APPROVED** [review]

**Pull request merged**